### PR TITLE
A number of testing and portability improvements.

### PR DESCRIPTION
**Pull request by nedbat**
I was working during the Open edX hackathon on adding tests to get ready for Python 3.  I also fixed a few Python 3 issues along the way.

### CODE REVIEW

**Comment by nedbat** [comment]
@efischer19 @robrap @doctoryes Not sure who might want to take a look.

File: i18n/validate.py [hunk 1]
```diff
@@ -81,7 +82,12 @@ def is_linguistic_tag(tag):
 
 def astral(msg):
     """Does `msg` have characters outside the Basic Multilingual Plane?"""
-    return any(ord(c) > 0xFFFF for c in msg)
+    # Python2 narrow builds present astral characters as surrogate pairs.
+    # By encoding as utf32, and decoding DWORDS, we can get at the real code
+    # points.
+    utf32 = msg.encode("utf32")[4:]         # [4:] to drop the bom
+    code_points = struct.unpack("%dI" % (len(utf32) / 4), utf32)
+    return any(cp > 0xFFFF for cp in code_points)
```
Comment (efischer19) [inline i18n/validate.py]: 😮 This is cool

**Comment by efischer19** [comment]
👍 Looks good! I hadn't heard of tox before, it seems really cool

**Comment by nedbat** [comment]
@bradenmacdonald keeping you up to date also

**Comment by doctoryes** [comment]
👍

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,29 @@
+[tox]
+envlist = py{27,35},coverage
+
+[testenv]
+deps =
+    coveralls
+    ddt
+    django>=1.8,<1.9
```
Comment (sarina) [inline tox.ini]: FYI, my changes that we needed to run on django 1.9 just got merged into the most recent version of markey, which is a dependency of some other thing we use that I don't recall, anyway point being you can probably start testing i18n-tools on django 1.9 now.

**Pull request merged**